addons/website_event_questions/models/event_question.py

```diff
@@ -25,24 +28,24 @@ def _constrains_event(self):
         if any(question.event_type_id and question.event_id for question in self):
             raise UserError(_('Question cannot belong to both the event category and itself.'))
 
-    @api.model
-    def create(self, vals):
-        event_id = vals.get('event_id', False)
-        if event_id:
-            event = self.env['event.event'].browse([event_id])
-            if event.event_type_id.use_questions and event.event_type_id.question_ids:
-                vals['answer_ids'] = vals.get('answer_ids', []) + [(0, 0, {
-                    'name': answer.name,
-                    'sequence': answer.sequence,
-                }) for answer in event.event_type_id.question_ids.filtered(lambda question: question.title == vals.get('title')).mapped('answer_ids')]
-        return super(EventQuestion, self).create(vals)
-
-
-class EventAnswer(models.Model):
-    _name = 'event.answer'
+
+class EventQuestionAnswer(models.Model):
+    """ Contains suggested answers to a 'simple_choice' event.question. """
+    _name = 'event.question.answer'
     _order = 'sequence,id'
-    _description = 'Event Answer'
+    _description = 'Event Question Answer'
 
     name = fields.Char('Answer', required=True, translate=True)
     question_id = fields.Many2one('event.question', required=True, ondelete='cascade')
     sequence = fields.Integer(default=10)
+
+class EventUserInputLine(models.Model):
+    """ Represents the user input answer for a single event.question """
+    _name = 'event.user_input.line'
```

Comment:
I would name it event.registration.answer, to stay in the event.registration namespace. Moreover we don't duplicate survey naming :D .